prbt_hardware_support/test/unittests/unittest_sto_modbus_adapter.cpp

```diff
@@ -444,6 +471,54 @@ TEST_F(PilzStoModbusAdapterTest, testNoSto)
   BARRIER("halt_callback");
 }
 
+/**
+ * @brief Helper class which allows us to test a function which is normally
+ * protected and, therefore, not directly accessible for testing.
+ */
+class TestingAdapterSto : public AdapterSto
+{
+public:
+  TestingAdapterSto(ros::NodeHandle& nh)
+    : AdapterSto(nh)
+  {}
+
+  virtual ~TestingAdapterSto() = default;
+
+public:
+  void testingUpdateSto(const bool sto)
+  {
+    updateSto(sto);
+  }
+};
+
+/**
+ * @brief Tests that unhold, recover, halt, hold services are ONLY called
+ * if the STO changes its value.
+ */
+TEST_F(PilzStoModbusAdapterTest, testSameStoValue)
+{
+  manipulator_.advertiseServices(nh_, HOLD_SERVICE_T, UNHOLD_SERVICE_T, HALT_SERVICE_T, RECOVER_SERVICE_T);
+
+  TestingAdapterSto sto_adapter(nh_);
+  {
+    InSequence dummy;
+    EXPECT_CALL(manipulator_, unholdCb(_,_)).Times(1);
+    EXPECT_CALL(manipulator_, recoverCb(_,_)).Times(1);
+    EXPECT_CALL(manipulator_, holdCb(_,_)).Times(0);
+    EXPECT_CALL(manipulator_, haltCb(_,_)).Times(0);
+  }
+  sto_adapter.testingUpdateSto(true);
+
+  {
+    InSequence dummy;
```

Comment:
I think the order of "Times(0)" expectations is redundant.

Reply:
Done.